text/XXXX-replace_publish_clone_with_push_pull.md

```diff
@@ -0,0 +1,258 @@
+
+### Apply `acceptSizeMax` to a history instead of single verisons
+When pushing a dataset, the receiver is in charge of what they will accept. A remote can reject a push request for any reason. One of the primary reasons for rejecting a push is do to size constraints. Qri has a configuration value `remote.acceptSizeMax` that currently polices the maximium size of a dataset version. Any dataset version that exeeds this value is rejected by default.
+
+This RFC proposes changing the `remote.acceptSizeMax` value in configuration to apply to the on-disk size of _the entire dataset history_. As an example: if `acceptSizeMax` is `300Mb` that quota can be filled with one version that is `300Mb` in size, three totally unique `100Mb` versions, or any other combination where the deduplicated size of all versions is below the maximum threshold. 
+
+We say totally unique versions because qri de-duplicates versioned data. A block of dataset data may be used in any number of versions, and should only count towards the size threshold once.
+
+### Retention Strategies
+This RFC introduces the term _retention strategy_ to describe the approach a peer takes to pruning versions when `acceptSizeMax` is exceeded. A few potential strategies, and their names might be:
+
+| name             | behavior                                                  |
+| ---------------- | --------------------------------------------------------- |
+| `none`           | error when quota is filled                                |
+| `archival`       | retain from initial commit forward, error if chain of versions is broken |
+| `rolling-window` | keep the latest, drop older versions                      |
+
+Defining possible retention strategies is beyond the scope of this RFC, but we can define the term here, and some of the problems retention strategies both create and address. It's also worth noting that retention strategies apply to _both_ pulling and pushing.
+
+In discussions we've come to realize that retention strategies are a major concern for the health of the qri network as a whole. For example: if all peers default to rolling-window strategy, we end up with a network of peers that "forget" early versions when the max size is exceeded. This destroys the auditability of a dataset, because no one retains the history. 
+
+A _retention_ strategy is also a _replication_ strategy. Put another way: which versions peers choose to keep determines the versions availabile to the network.
+
+### Remotes default to a `none` retention strategy
+If the user requests to push N versions for Y data size, the typical response is to accept all pushed verions, so long as the total disk space consumed is less than `remote.acceptSizeMax`. When a user tries to push a number of versions above the max,  `__ Mb` over the bar
```

Comment:
unfinished, I believe. What happens when a user tries to push a number of versions above the max? Error response and no pushing actually occurs?